src/data_distribution/first_touch_distribution.hpp

```diff
@@ -11,6 +11,27 @@
 
 #include "base_distribution.hpp"
 
+/**
+ * @brief The number of std::size_t's used to represent a single entry
+ * in the global_owners_dir
+ */
+constexpr std::size_t single_entry_size = 1;
+
+/**
+ * @brief The number of single_entry used to classify an ArgoDSM page's
+ * ownership in the global_owners_dir
+ * @note This number derives from the definition of the global_owners_dir
+ * in the MPI backend. Changes to this must be applied in both locations
+ * and should probably be centralized at some point.
+ */
+constexpr std::size_t page_info_size = single_entry_size * 3;
+
+/** @brief This constant is used to represent an invalid node id */
+constexpr argo::node_id_t invalid_node_id = -1;
+
+/** @brief This constant is used to represent an invalid offset */
+constexpr std::size_t invalid_offset = SIZE_MAX;
```

Comment:
both of these sound like they belong to either global_ptr.hpp or base_distribution.hpp. Is there a good reason to have them defined only for first_touch?